### DISCUSSION TITLE
BitwardenRs won't install on Raspberry Pi 3+

### DISCUSSION THREAD

**Discussion by ghost** [dani-garcia/vaultwarden#1345] [Q&A]
Hello, 

Whenever I run the "docker pull bitwardenrs/server:raspberry" command this will happen:
"Error response from daemon: manifest for bitwardenrs/server:raspberry not found: manifest unknown: manifest unknown".

I followed this installation guide: https://www.youtube.com/watch?v=eCJA1F72izc

I would be very thankful if someone could help me.

**Comment by BlackDex** [comment]
That guide is using an outdated tag.
You can just do a `docker pull bitwardenrs/server` and it should work.

**Reply by ghost** [reply]
But when I do it this way it uses the Tag latest automatically and then there is an error which says that this version is Not Right for the ARM Architecture

**Reply by BlackDex** [reply]
That is a know issues which also has been fixed but not yet in the `latest` release.
It is fixed in the `testing` release.

Also, that message does not matter, the container will work.

**Reply by 1011001-p** [reply]
I´m having exactly the same issue as @Techsized but on a Raspberry Pi 4 Model B Rev 1.2 and if I ignore the warning the container does not work: is there a fix for this?

**Comment by BlackDex** [comment]
Which version of docker are you using?
And what is the exact message you get when you try.

`docker run --rm -it bitwardenrs/server:testing`

**Reply by 1011001-p** [reply]
I´m currently trying with the lastest (Docker version 20.10.3, build 48d30b5)
But some hours ago I tried installing an older version of Docker with the following command that I found [here](https://github.com/portainer/portainer/issues/4585#issuecomment-751729241):
`curl -sSL https://get.docker.com | sed 's/docker-ce/docker-ce=5:19.03.13~3-0~raspbian-buster/' | sh`

**The command you mentioned seems to work: but I´m looking for a stable release if possible.**
_______
_Here is the output of what happens when I try to run the lastest version (on Docker version 20.10.3, build 48d30b5):_

`sudo docker run -d --name bitwarden --restart unless-stopped -v /mnt/bitwarden/:/data/ -p 1999:80 bitwardenrs/server`
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm/v7) and no specific platform was requested
744c6ed49350719e445696de701ef391ec81432bf9b3263acd5414a9d248739f
`docker ps`
CONTAINER ID   IMAGE                COMMAND                  CREATED          STATUS                           PORTS     NAMES
744c6ed49350   bitwardenrs/server   "/usr/bin/entry.sh /…"   12 seconds ago   Restarting (101) 2 seconds ago             bitwarden
_______
If I do docker ps it restarts itself indefinetly: If I run it without the `--restart unless-stopped` flag it does not appear on the `docker ps` command.

**Comment by BlackDex** [comment]
And what does `docker logs bitwarden` tell you?
Also, please try the command above i mentioned instead of the command you tried.

I just tested this on my RPI 2b and both latest and testing work.
And latest does give me a message regarding the arch but it does still start and work.

**Reply by 1011001-p** [reply]
These are the logs when I run `sudo docker run -d --name bitwarden -v /mnt/bitwarden/:/data/ -p 1999:80 bitwardenrs/server`

/--------------------------------------------------------------------\
|                       Starting Bitwarden_RS                        |
|                           Version 1.18.0                           |
|--------------------------------------------------------------------|
| This is an *unofficial* Bitwarden implementation, DO NOT use the   |
| official channels to report bugs/features, regardless of client.   |
| Send usage/configuration questions or feature requests to:         |
|   https://bitwardenrs.discourse.group/                             |
| Report suspected bugs/issues in the software itself at:            |
|   https://github.com/dani-garcia/bitwarden_rs/issues/new           |
\--------------------------------------------------------------------/

[INFO] No .env file found.

[2021-02-06 22:25:19.135][panic][ERROR] thread 'main' panicked at 'Failed to turn on WAL: DatabaseError(__Unknown, "database is locked")': src/db/mod.rs:275
   0: backtrace::backtrace::trace
   1: backtrace::capture::Backtrace::new
   2: bitwarden_rs::init_logging::{{closure}}
   3: std::panicking::rust_panic_with_hook
             at rustc/da384694807172f0ca40eca2e49a11688aba6e93/library/std/src/panicking.rs:595:17
   4: std::panicking::begin_panic_handler::{{closure}}
             at rustc/da384694807172f0ca40eca2e49a11688aba6e93/library/std/src/panicking.rs:497:13
   5: std::sys_common::backtrace::__rust_end_short_backtrace
             at rustc/da384694807172f0ca40eca2e49a11688aba6e93/library/std/src/sys_common/backtrace.rs:141:18
   6: rust_begin_unwind
             at rustc/da384694807172f0ca40eca2e49a11688aba6e93/library/std/src/panicking.rs:493:5
   7: core::panicking::panic_fmt
             at rustc/da384694807172f0ca40eca2e49a11688aba6e93/library/core/src/panicking.rs:92:14
   8: core::option::expect_none_failed
             at rustc/da384694807172f0ca40eca2e49a11688aba6e93/library/core/src/option.rs:1268:5
   9: bitwarden_rs::db::sqlite_migrations::run_migrations
  10: bitwarden_rs::db::DbPool::from_config
  11: bitwarden_rs::util::retry_db
  12: bitwarden_rs::main
  13: std::sys_common::backtrace::__rust_begin_short_backtrace
  14: std::rt::lang_start::{{closure}}
  15: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once
             at rustc/da384694807172f0ca40eca2e49a11688aba6e93/library/core/src/ops/function.rs:259:13
      std::panicking::try::do_call
             at rustc/da384694807172f0ca40eca2e49a11688aba6e93/library/std/src/panicking.rs:379:40
      std::panicking::try
             at rustc/da384694807172f0ca40eca2e49a11688aba6e93/library/std/src/panicking.rs:343:19
      std::panic::catch_unwind
             at rustc/da384694807172f0ca40eca2e49a11688aba6e93/library/std/src/panic.rs:396:14
      std::rt::lang_start_internal
             at rustc/da384694807172f0ca40eca2e49a11688aba6e93/library/std/src/rt.rs:51:25
  16: main
  17: __libc_start_main

----

**If I run the command you provided it seems to work: but I´m looking for a stable release if possible**. The output by terminal is the following:

/--------------------------------------------------------------------\
|                       Starting Bitwarden_RS                        |
|                      Version 1.18.0-85e3c735                       |
|--------------------------------------------------------------------|
| This is an *unofficial* Bitwarden implementation, DO NOT use the   |
| official channels to report bugs/features, regardless of client.   |
| Send usage/configuration questions or feature requests to:         |
|   https://bitwardenrs.discourse.group/                             |
| Report suspected bugs/issues in the software itself at:            |
|   https://github.com/dani-garcia/bitwarden_rs/issues/new           |
\--------------------------------------------------------------------/

[INFO] No .env file found.

[2021-02-06 22:28:13.957][bitwarden_rs][INFO] JWT keys don't exist, checking if OpenSSL is available...
OpenSSL 1.1.1d  10 Sep 2019
[2021-02-06 22:28:13.970][bitwarden_rs][INFO] OpenSSL detected, creating keys...
Generating RSA private key, 2048 bit long modulus (2 primes)
..............................................+++++
...................................+++++
e is 65537 (0x010001)
writing RSA key
writing RSA key
[2021-02-06 22:28:15.839][bitwarden_rs][INFO] Keys created correctly.
Running migration 20180114171611
Running migration 20180217205753
Running migration 20180427155151
Running migration 20180508161616
Running migration 20180525232323
Running migration 20180601112529
Running migration 20180711181453
Running migration 20180827172114
Running migration 20180910111213
Running migration 20180919144557
Running migration 20181127152651
Running migration 20190526216651
Running migration 20191010083032
Running migration 20191117011009
Running migration 20200313205045
Running migration 20200409235005
Running migration 20200701214531
Running migration 20200802025025
Running migration 20201130224000
Running migration 20201209173101
[2021-02-06 22:28:16.668][start][INFO] Rocket has launched from http://0.0.0.0:80

**Comment by BlackDex** [comment]
You seem to be using a shared filesystem, probably cifs/samba.
If you want to use this you need to disable WAL:
https://github.com/dani-garcia/bitwarden_rs/blob/85e3c73525d327042c1ad142e48c044a5dbdd89c/.env.template#L87

Or use an other database like MariaDB or Postgresql.

**Reply by 1011001-p** [reply]
Yes, I´m using CIFS: my idea was to save the whole bitwarden folder into a share I created on my Synology NAS: I didn´t know it could cause issues.
**I changed the directory to a local one and it works! Thank you! I have been trying to do this for like 8 hours now!**
